### ISSUE TITLE
[Feature Request] 3D Gaussian Splatting MCMC support

### ISSUE THREAD

**Issue by equilento** [playcanvas/engine#6817]
It would be awesome if Markov Chain Monte Carlo is supported. PostShot just added support and the results are MUCH better. I actually tried loading MCMC ply file inside SuperSplat but it doesn't really show anything, just what seems a bunch of very distant splats. Tried compressing it and loading inside Playcanvas and it does in fact load but it looks like a complete mess.

**Comment by slimbuck** [comment]
Hi @equilento,

From a quick search I wasn't able to find an example file for MCMC GS.

If you are able to provide an example file then I can investigate the issue.

Thanks!

**Comment by equilento** [comment]
Sure, thanks! here is an example file:
https://easyupload.io/rh2nww

**Comment by slimbuck** [comment]
Hi @equilento,

This file is actually loading fine. The issue is there are very large splats far from the center of the scene. (So the camera is initialised far away to frame everything).

Here is a quick video showing how I select the very far away splats using the new DATA panel:

https://github.com/user-attachments/assets/09eebf17-022f-4939-a2d8-d6e3404fbd56

After deleting the large splats the scene flickers still in supersplat. I will fix this issue next.

**Comment by equilento** [comment]
Yeah, not only does it flicker but it is a mess. This is another model MCMC and "regular" GS in comparison.
<img width="743" alt="ADC Splat" src="https://github.com/user-attachments/assets/ee3e4327-f735-4134-be1e-299ccaf45672">
<img width="966" alt="MCMC Splat" src="https://github.com/user-attachments/assets/c9ceb256-f732-4879-bd7d-6f55d38e87dc">

**Comment by slimbuck** [comment]
Hi @equilento,

Sorry I don't understand what is being shown in the screenshot above.

Is the bottom image an MCMC scene loaded in Supersplat or...?

**Comment by equilento** [comment]
Yes, the bottom one is MCMC scene after I did the clean up like you showed. I tried loading it inside the editor and the results are the same.

**Comment by slimbuck** [comment]
Ah, right. Can you send me the original GS scene and instructions on how to replicate this issue?

**Comment by equilento** [comment]
Ok so I stumbled on something quite interesting. After I deleted far-away unnecessary splats like you suggested, focused on the model and zoomed in I get something that looks very bad (fig. 1) BUT after I then compress it, save it and open it again inside SuperSplat I get a model that looks much better (fig. 2) but still not as good as it is supposed to look like (fig. 3)
<img width="849" alt="fig  1" src="https://github.com/user-attachments/assets/8619c0d0-2f3c-4708-b249-8cfcf572e4cd">
<img width="897" alt="fig  2" src="https://github.com/user-attachments/assets/6bbb2449-486a-4194-bfcb-05878b18785e">
<img width="839" alt="fig  3" src="https://github.com/user-attachments/assets/8cbc8833-8f64-4346-bc05-3cc597a18b00">

Uncompressed/unedited file:
https://easyupload.io/gdbt1n

**Comment by slimbuck** [comment]
Hi @equilento,

Thanks for this!

The issue with the flickering is that deleted splats are not actually removed from the scene, they are just made transparent. This means the sorter buckets still include all splats, even though they're deleted. (We perform a very fast bucket sort for performance reasons).

I am busy updating the engine to support quickly removing deleted splats from sorting and rendering. This removers the flickering (equivalent of saving and reloading the scene) and speeds up the scene as you edit.

I will double check with the scene you gave me, but seems like it's the same issue.

Thanks again!

**Comment by slimbuck** [comment]
BTW for your testing, it's probably better to save the cut-down scene and then load that instead of saving a compressed version of the scene (since compression is lossy). Only the save the compressed version for runtime use.

**Comment by equilento** [comment]
I tried cutting down the original, saved the uncompressed file, reloading and then compressing it, opened it and it seems fine now. Couple of days ago I tried just compressing the original and I put it inside PlayCanvas editor and the flickering was present, even though I haven't done any editing. Why would that be? If I understood you correctly this shouldn't happen. Anyway this seems like a quick fix for me for now. Don't know how this would be reflected on bigger scenes.

**Comment by mvaligursky** [comment]
@slimbuck - can this be closed or is more work needed?

**Comment by willeastcott** [comment]
That's a yes. 🚀

**Comment by willeastcott** [comment]
Ooops. I should let the PR close this. 😊
